### PR TITLE
Automatic schematic net labeling for passive-chip for complex traces

### PR DESCRIPTION
**Pull request by AnasSarkiz**
/claim #387

### CODE REVIEW

**Review by seveibar — COMMENTED** [review]
Can we remove code related to the sides, just to simplify the files

File: lib/components/primitive-components/Trace/Trace.ts [hunk 1]
```diff
@@ -892,5 +939,60 @@ export class Trace
...
+        db.schematic_net_label.insert({
+          text: fullLabel,
+          source_net_id: passiveSidePosition.port.source_port_id!,
+          anchor_position: passiveSidePosition.position,
+          center: passiveSidePosition.position,
+          anchor_side:
+            getEnteringEdgeFromDirection(
+              passiveSidePosition.facingDirection!,
+            ) ?? "bottom",
+        })
```
Comment (seveibar) [inline lib/components/primitive-components/Trace/Trace.ts:990]: Use the existing function for creating the net labels, just modify it to derive the net label name if it doesnt have schDisplayLabel

Dont allow the schematic trace to be inserted if its complex

Moving the counting function outside this class, its not stateful, you give it the path and it returns the complexElementCount

**Review by seveibar — CHANGES_REQUESTED** [review]

File: lib/components/primitive-components/Trace/Trace.ts (hunk 1 above)
Comment (AnasSarkiz) [reply lib/components/primitive-components/Trace/Trace.ts:990]: modify the `_doInitialSchematicTraceRenderWithDisplayLabel` method and add the logic to derive the netlabel name if it doesn't have `schDisplayLabel`
Screenshot 2024-12-08 at 9 03 45 AM

File: lib/components/primitive-components/Trace/Trace.ts [hunk 2]
```diff
@@ -658,18 +659,33 @@ export class Trace
...
+      firstPort.parent?.config.shouldRenderAsSchematicBox
+    const isSecondPortSchematicBox =
+      secondPort.parent?.config.shouldRenderAsSchematicBox
+    const firstPortName = isFirstPortSchematicBox
+      ? firstPort?.props.name
+      : `${secondPort?.parent?.props.name}_${secondPort?.props.name}`
+    const secondPortName = isSecondPortSchematicBox
+      ? secondPort?.props.name
+      : `${firstPort?.parent?.props.name}_${firstPort?.props.name}`
+
```
Comment (seveibar) [inline lib/components/primitive-components/Trace/Trace.ts:686]: The names should always be identical (its just a schematic convention for net labels), if both parents are schematic boxes we could do a joint name (“U1_SDA_U2_SCL”)

i know thats pretty gross/long but i cant think of a better solution for two boxes at the moment.

**Review by seveibar — COMMENTED** [review]
AWESOME, but the matching labels is important, i was co fused pokking at the snapshot with PWR1 because the net labels dont match and that is essential

File: lib/components/primitive-components/Trace/Trace.ts (hunk 2 above)
Comment (AnasSarkiz) [reply lib/components/primitive-components/Trace/Trace.ts:686]: @seveibar  The code is currently functioning only for passive-to-chip connections, as indicated by the issue. Should I modify it to handle all trace types, including passive-to-passive and chip-to-chip connections as well?

Comment (AnasSarkiz) [reply lib/components/primitive-components/Trace/Trace.ts:686]: @seveibar

**Comment by AnasSarkiz** [comment]
> AWESOME, but the matching labels is important, i was confused poking at the snapshot with PWR1 because the net labels dont match and that is essential

@seveibar 
Screenshot 2024-12-09 at 6 17 01 PM

the name of the port is `pin2` with aliases `FB` maybe be we need to make some changes

Screenshot 2024-12-09 at 6 15 14 PM
Screenshot 2024-12-09 at 6 15 26 PM

**Comment by seveibar** [comment]
@AnasSarkiz no the issue is one is `PWR_pin2` and the other is `pin2`, they should both be `PWR_pin2`

File: lib/components/primitive-components/Trace/Trace.ts (hunk 2 above)
Comment (seveibar) [reply lib/components/primitive-components/Trace/Trace.ts:686]: no passive to chip is fine to start!

Comment (seveibar) [reply lib/components/primitive-components/Trace/Trace.ts:686]: but the labels should be the same on each side!

**Comment by AnasSarkiz** [comment]
@seveibar 
now, both sides have the same name of the netlabel


Screenshot 2024-12-10 at 5 33 41 PM

**Pull request merged**